scuba/config.py

```diff
@@ -65,7 +91,31 @@ def from_yaml(self, node):
             raise yaml.YAMLError('Key "{}" not found in {}'.format(key, filename))
         return cur
 
+    def override(self, node):
+        '''
+        Implements !override constructor
+        '''
+        # Load the content from the node, as a scalar
+        content = self.construct_scalar(node)
+
+        # Dynamically add an OverrideMixin to the resulting object's type
+        obj = yaml.load(content, lambda s: Loader(s, root=self._root))
+        if obj is None:
+            obj = OverrideNone()
+        else:
+            objtype = type(obj)
+            mixin_type = type('Override' + objtype.__name__, (objtype, OverrideMixin), dict())
+
+            try:
+                obj.__class__ = mixin_type
+            except TypeError:
+                # Primitive classes (e.g., int, str) don't support __class__ assignment
+                obj = mixin_type(obj)
+
+        return obj
+
 Loader.add_constructor('!from_yaml', Loader.from_yaml)
+Loader.add_constructor('!override', Loader.override)
```

Comment:
Should we be concerned about, or provide protections against, users using `!override` in other, inappropriate contexts?

Reply:
I don't think so. There are no protections against using the many tags built-in to PyYAML (see [here](https://pyyaml.org/wiki/PyYAMLDocumentation#yaml-tags-and-python-types)) in inappropriate contexts. Why should application-specific tags behave differently?

I think the only possible answer to this is that the `!override` will be silently ignored (and behave in ways the user doesn't expect), while using most of the other possible inappropriate tags would likely cause an error. However, It doesn't seem sustainable to make everything that accesses the YAML check for an `OverrideMixin` (and then raise an error), and building knowledge of which keys support `!override` into the `Loader` seems like misplaced logic.